templates/types/streaming/nextjs/app/api/chat/route.ts

```diff
@@ -54,6 +60,17 @@ export async function POST(request: NextRequest) {
       data?.imageUrl,
     );
 
+    // Init Vercel AI StreamData
+    const vercelStreamData = new StreamData();
+
+    // Setup callback for streaming data before chatting
+    Settings.callbackManager = new CallbackManager({
```

Comment:
@thucpn can you try this (`onRetrieve` will be deprecated soon)
```suggestion
    Settings.callbackManager.on("retrieve"
```